rmf_demos_panel/rmf_demos_panel/static/src/components/tasks/task-card.tsx

```diff
@@ -24,82 +26,123 @@ interface TaskCardProps {
+          <CardContent>
+            <Grid container>
+              <Grid item xs={12}>
+                {returnProgressBar(currentTaskStatus)}
               </Grid>
-            </CardContent>
-        </Card>
-    );                              
+                <Grid item xs={6}>
+                  <Typography variant="subtitle2" align="left" color="textSecondary" gutterBottom>
+                    Task ID
+                  </Typography>
+                </Grid>
+                <Grid item xs={6}><Typography>{taskState.task_id}</Typography></Grid>
+                <Grid item xs={6}>
+                  <Typography variant="subtitle2" align="left" color="textSecondary" gutterBottom>
+                    Details
+                  </Typography>
+                </Grid>
+                <Grid item xs={6}><Typography>{taskState.description}</Typography></Grid>
+                <Grid item xs={6}>
+                  <Typography variant="subtitle2" align="left" color="textSecondary" gutterBottom>
+                    Robot
+                  </Typography>
+                </Grid>
+                <Grid item xs={6}><Typography>{taskState.robot_name}</Typography></Grid>
+                <Grid item xs={6}>
+                  <Typography variant="subtitle2" align="left" color="textSecondary" gutterBottom>
+                    Task Type
+                  </Typography>
+                </Grid>
+                <Grid item xs={6}><Typography>{taskState.task_type}</Typography></Grid>
+                <Grid item xs={6}>
+                  <Typography variant="subtitle2" align="left" color="textSecondary" gutterBottom>
+                    Priority
+                  </Typography>
+                </Grid>
+                <Grid item xs={6}><Typography>{taskState.priority}</Typography></Grid>
+                <Grid item xs={6}>
+                  <Typography variant="subtitle2" align="left" color="textSecondary" gutterBottom>
+                    Task State
+                  </Typography>
+                </Grid>
+                <Grid item xs={6}>
+                  <StyledChip state={taskState.state}/>
+                </Grid>
+                <Grid item xs={3}>
+                  <Typography variant="subtitle2" align="left" color="textSecondary" gutterBottom>
+                    Start:
+                  </Typography>
+                </Grid>
+                <Grid item xs={3}><Typography>{taskState.start_time}</Typography></Grid>
+                <Grid item xs={3}>
+                  <Typography variant="subtitle2" align="left" color="textSecondary" gutterBottom>
+                    End:
+                  </Typography>
+                </Grid>
+                <Grid item xs={3}><Typography>{taskState.end_time}</Typography></Grid>
+                <Grid item xs={12}><Button variant="outlined" onClick={() => handleCancel(taskState.task_id)} disabled={ currentTaskStatus === 'Cancelled' ? true : false }>Cancel Task</Button></Grid>
```

Comment:
It will be nice if we can blur out the cancel button for `Failed` and `Completed` task card. Not sure if the fix should be here.